statsmodels/stats/proportion.py

```diff
@@ -123,6 +123,183 @@ def func(qi):
+                             chi2 * (1 - estimated_parameters))
+        region = ((2 * n * estimated_parameters + chi2 +
+                   np.array([- np.sqrt(delta), np.sqrt(delta)])) /
+                  (2 * (chi2 + n))).T
+    elif method == 'sison_glaz':
+        # Define a few functions we'll use a lot.
+        def poisson_interval(interval, l):
+            """Compute P(b <= Z <= a) where Z ~ Poisson(l) and
+            `interval = (b, a)`."""
+            b, a = interval
+            return stats.poisson.cdf(a, l) - stats.poisson.cdf(b - 1, l)
+
+        def truncated_poisson_factorial_moment(interval, r, l):
+            """Compute mu_r, the r-th factorial moment of a poisson random
+            variable of parameter `l` truncated to `interval = (b, a)`."""
+            b, a = interval
+            return l ** r * (1 - ((poisson_interval((a - r + 1, a), l) -
+                                   poisson_interval((b - r, b - 1), l)) /
+                                  poisson_interval((b, a), l)))
+
+        def edgeworth(intervals):
+            """Compute the Edgeworth expansion term of Sison & Glaz's formula
+            (1) (approximated probability for multinomial proportions in a
+            given box)."""
+            # Compute means and central moments of the truncated poisson
+            # variables.
+            mu_r1, mu_r2, mu_r3, mu_r4 = [
+                np.array([truncated_poisson_factorial_moment(interval, r, l)
+                          for (interval, l) in zip(intervals, counts)])
+                for r in range(1, 5)
+            ]
+            mu = mu_r1
+            mu2 = mu_r2 + mu - mu ** 2
+            mu3 = mu_r3 + mu_r2 * (3 - 3 * mu) + mu - 3 * mu ** 2 + 2 * mu ** 3
+            mu4 = (mu_r4 + mu_r3 * (6 - 4 * mu) +
+                   mu_r2 * (7 - 12 * mu + 6 * mu ** 2) +
+                   mu - 4 * mu ** 2 + 6 * mu ** 3 - 3 * mu ** 4)
+
+            # Compute expansion factors, gamma_1 and gamma_2.
+            g1 = mu3.sum() / mu2.sum() ** 1.5
+            g2 = (mu4.sum() - 3 * (mu2 ** 2).sum()) / mu2.sum() ** 2
+
+            # Compute the expansion itself.
+            x = (n - mu.sum()) / np.sqrt(mu2.sum())
+            phi = np.exp(- x ** 2 / 2) / np.sqrt(2 * np.pi)
+            H3 = x ** 3 - 3 * x
+            H4 = x ** 4 - 6 * x ** 2 + 3
+            H6 = x ** 6 - 15 * x ** 4 + 45 * x ** 2 - 15
+            f = phi * (1 + g1 * H3 / 6 + g2 * H4 / 24 + g1 ** 2 * H6 / 72)
+            return f / np.sqrt(mu2.sum())
+
+        def approximated_multinomial_interval(intervals):
+            """Compute approximated probability for
+            Multinomial(n, estimated_parameters) to be in `intervals`
+            (Sison & Glaz's formula (1))."""
+            return (np.prod([poisson_interval(interval, l) for (interval, l)
+                             in zip(intervals, counts)]) *
+                    edgeworth(intervals) / stats.poisson.pmf(n, n))
```

Comment:
written this way it looks likes product and division
I would expect that using exp sum log would be numerically more stable.